x-pack/test/functional/apps/lens/add_to_dashboard.ts

```diff
@@ -0,0 +1,257 @@
+          await PageObjects.lens.goToTimeRange();
+
+          await PageObjects.lens.configureDimension({
+            dimension: 'lnsXY_yDimensionPanel > lns-empty-dimension',
+            operation: 'avg',
+            field: 'bytes',
+          });
+
+          await PageObjects.lens.switchToVisualization('lnsMetric');
+
+          await PageObjects.header.waitUntilLoadingHasFinished();
+          await PageObjects.lens.assertMetric('Average of bytes', '5,727.322');
+
+          await PageObjects.header.waitUntilLoadingHasFinished();
+          await testSubjects.click('lnsApp_saveButton');
+
+          const hasOptions = await testSubjects.exists('add-to-dashboard-options');
+          expect(hasOptions).to.eql(false);
+        });
+      });
+
+      describe('dashboard read-only privileges', () => {
+        before(async () => {
+          await security.testUser.setRoles(
+            ['test_logstash_reader', 'global_visualize_all', 'global_dashboard_read'],
+            true
+          );
+        });
+
+        after(async () => {
+          await security.testUser.restoreDefaults();
+        });
+
+        it('should not display dashboard flow prompt', async () => {
+          await PageObjects.common.navigateToApp('visualize');
+          await PageObjects.header.waitUntilLoadingHasFinished();
+          await PageObjects.visualize.gotoLandingPage();
+
+          const hasPrompt = await testSubjects.exists('visualize-dashboard-flow-prompt');
+          expect(hasPrompt).to.eql(false);
+        });
+
+        it('should only display "Existing" add-to-dashboard option', async () => {
+          await PageObjects.visualize.navigateToNewVisualization();
+          await PageObjects.visualize.clickVisType('lens');
+          await PageObjects.lens.goToTimeRange();
+
+          await PageObjects.lens.configureDimension({
+            dimension: 'lnsXY_yDimensionPanel > lns-empty-dimension',
+            operation: 'avg',
+            field: 'bytes',
+          });
+
+          await PageObjects.lens.switchToVisualization('lnsMetric');
+
+          await PageObjects.header.waitUntilLoadingHasFinished();
+          await PageObjects.lens.assertMetric('Average of bytes', '5,727.322');
+
+          await PageObjects.header.waitUntilLoadingHasFinished();
+          await testSubjects.click('lnsApp_saveButton');
+
+          const hasOptions = await testSubjects.exists('add-to-dashboard-options');
+          expect(hasOptions).to.eql(true);
+
+          const hasCreateNewOption = await find.existsByCssSelector(
+            'input[id="new-dashboard-option"]'
+          );
+          const hasExistingOption = await find.existsByCssSelector(
+            'input[id="existing-dashboard-option"]'
+          );
+          const hasAddToLibraryOption = await find.existsByCssSelector(
+            'input[id="add-to-library-option"]'
+          );
+
+          expect(hasCreateNewOption).to.eql(false);
+          expect(hasExistingOption).to.eql(true);
```

Comment:
I know this is just the test PR, but isn't this a bug (or at least confusing behavior)? If dashboard is set to "read", why is it possible to add panels to existing dashboards from within Lens? In dashboard we don't allow to enter "Edit" mode, so it's weird to allow it from this direction. Just IMHO, please disregard if this was discussed.

Reply:
I double checked, and it seems like this should be changed to hide all dashboard options if the user has read-only permissions for dashboard. Currently with read-only adding a new vis to an existing dashboard will add it in view mode, which is a really strange experience.

Reply:
👍